### PR TITLE
[es6-bookreader-core] extract BookReader.util

### PR DESCRIPTION
**Pull request by cdrini**
Extract a chunk of BookReader.js: static util methods. Turns out most of the jest tests were for this anyways :P

### CODE REVIEW

File: src/js/BookReader.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ This file is part of BookReader.
 
 */
 import { version as VERSION } from '../../package.json';
+import * as utils from './BookReader/utils.js';
```
Comment (iisa) [inline src/js/BookReader.js]: 👏

File: tests/BookReader/utils.test.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import sinon from 'sinon';
+import {
+    clamp,
+    cssPercentage,
+    debounce,
+    decodeURIComponentPlus,
+    encodeURIComponentPlus,
+    escapeHTML,
+} from '../../src/js/BookReader/utils.js';
```
Comment (iisa) [inline tests/BookReader/utils.test.js]: Q: is this every util?

File: tests/BookReader/utils.test.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+test('Encodes a URI component and converts emptyStr to +', () => {
+    expect(encodeURIComponentPlus("?x=test ")).toEqual("%3Fx%3Dtest+");
+    expect(encodeURIComponentPlus("ABC abc 123")).toEqual("ABC+abc+123");
+});
+
+let clock;
+
+beforeEach(() => {
+  clock = sinon.useFakeTimers();
```
Comment (iisa) [inline tests/BookReader/utils.test.js]: cool

File: tests/BookReader/utils.test.js (hunk 2 above)
Comment (cdrini) [reply tests/BookReader/utils.test.js]: All but 1 I think :P

File: tests/BookReader/utils.test.js (hunk 3 above)
Comment (cdrini) [reply tests/BookReader/utils.test.js]: This was here before me, so can't take full credit :P

**Comment by cdrini** [comment]
> I saw some string concatenations that were touched, but I am assuming that we will do that update for that in another round of fixes

Exactly; as I split things out, I'll es6-ify them :+1:

**Pull request merged**